pkg/database/reader.go

```diff
@@ -115,14 +115,8 @@ func (db *DatabaseImpl) QueryLogURL(ctx context.Context, kind, apiVersion, names
 func (db *DatabaseImpl) QueryResources(ctx context.Context, kind, apiVersion, namespace, name,
 	continueId, continueDate string, labelFilters *LabelFilters, limit int) ([]string, int64, string, error) {
 	sb := db.getSelector().ResourceSelector()
-	sb.Where(
-		db.filter.KindFilter(sb.Cond, kind),
-		db.filter.ApiVersionFilter(sb.Cond, apiVersion),
-	)
-
-	if namespace != "" {
-		sb.Where(db.filter.NamespaceFilter(sb.Cond, namespace))
-	}
+	mainWhereClause := db.mainWhereClause(kind, apiVersion, namespace)
+	sb.AddWhereClause(mainWhereClause)
```

Comment:
I don't think you need this. After the two removed `sb.Where` you can pass `sb.WhereClause` to the `NotEqualsLabelFilter`, this way you avoid the extraction to a different function

Reply:
I've refactored this piece and removed the extra method. Please take a look and tell me if that's what you meant